Closes #2527: Add badge view mode, update existing badge view mode to "linked badge" and setting terms up to use them.

trackleft opened the pull request:
## Description
Adding new view mode for taxonomy (Badge no link)
Adding new displays for `az_publication_categories`

Rename badge to linked badge

Possible todos:

- [ ] Add both displays to all vocabularies that have badge
- [ ] Add new display to az_publication, reference with tags
- [ ] Add new display to az_publication, reference with linked tags.

## Related issues
Closes #2527 

## How to test general term integration
Log in as admin
Navigate to /admin/structure/taxonomy/manage/az_page_categories/overview/display (Or any other taxonomy)
<img width="1659" alt="image" src="https://github.com/az-digital/az_quickstart/assets/1023167/1f3b6657-1145-4b81-9e41-95200b0ba51c">
You should now be able to enable the badge display (not a link)
Also you should see the new name of the existing badge display (linked badge)

## How to test publication term integration
Navigate to /admin/structure/taxonomy/manage/az_publication_categories/overview/display/az_badge
<img width="1662" alt="image" src="https://github.com/az-digital/az_quickstart/assets/1023167/8b40a49a-c7c2-46ca-9c71-ffc5ca9b40c8">


Navigate to /admin/structure/taxonomy/manage/az_publication_categories/overview/display/az_badge_no_link
<img width="1673" alt="image" src="https://github.com/az-digital/az_quickstart/assets/1023167/7984b16c-591a-42de-bf46-02e49bc33dba">


To enable this on an entity reference field in a view display for a fieldable entity, here are the settings you need.
<img width="1635" alt="image" src="https://github.com/az-digital/az_quickstart/assets/1023167/8c97265d-b425-4044-b28d-7804efb27c8f">

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
   - [ ] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
   - [ ] Adding experimental module
   - [ ] Update experimental module
- **Minor release changes**
   - [x] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Upgrade experimental module to stable
   - [ ] Enable existing module by default or database update
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- **Other or unknown**
   - [ ] Other or unknown

### Drupal core
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major or minor level update
- **Other or unknown**
   - [ ] Other or unknown

### Drupal contrib projects
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major level update
- **Other or unknown**
   - [ ] Other or unknown

## Checklist
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

trackleft committed:
Adding view mode and setting terms up to use them

trackleft committed:
Update core.entity_view_display.taxonomy_term.az_publication_categories.az_badge_no_link.yml

trackleft committed:
Merge branch 'main' into issue/2527-add-category-to-citation-ref-display

trackleft committed:
Merge branch 'main' into issue/2527-add-category-to-citation-ref-display

joshuasosa commented:
Interesting, you're changing Badge to Linked Badge and creating a new Badge that's not linked. But then az_badge is the linked one while a new az_badge_no_link is the new Badge that's not linked. IMO it'd be better to keep labeling consistent: az_badge is Badge that's not linked, and az_badge_linked is the linked one. Not an important suggestion if ignored though.

I see this is for az_publication_categories, but it would also be useful on az_person_categories. The sites I've worked with sometimes need categories for people, but they don't want those badges linking to those generic taxonomy pages. This issue was also brought up in [#azdigital-support](https://uarizona.slack.com/archives/C0SAW6H4J/p1680210998459259).

danahertzberg commented:
I agree I've heard most gripes about the badges being linked as an issue. I would just like to verify that the badge changes are not changing existing functionality, but only adding functionality.

I like Joshua's opinion on having naming consistency if possible.

trackleft committed:
Changing az_badge to not use a link, and az_badge_link to be a linked badge.

trackleft commented:
I've updated the naming conventions to reflect the most commonly requested configuration by default.
- az_badge (no longer a linked badge)
- changed `az_badge_no_link` to `az_badge_link`  which is now a linked badge.

trackleft committed:
Update existing view modes to remove link.

trackleft committed:
Update person primary categories

trackleft committed:
Change az_badge_link to az_badge_alt

trackleft commented on modules/custom/az_event/config/install/core.entity_view_display.taxonomy_term.az_event_categories.az_badge.yml:
```diff
@@ -13,23 +13,9 @@ third_party_settings:
-        id: ''
-        target: entity
-        custom_uri: ''
-        target_attribute: default
```
Now that I've updated the second view mode's id to `az_badge_alt` we probably don't need to update this.

trackleft committed:
Remove dependency on field_group module

danahertzberg commented on modules/custom/az_core/config/install/core.entity_view_mode.taxonomy_term.az_badge_alt.yml:
```diff
@@ -0,0 +1,9 @@
+  module:
+    - taxonomy
+id: taxonomy_term.az_badge_alt
+label: Alternate Badge
```
Is there a reason this is named "Alternate Badge" rather than "Linked Badge"? Are there other functionality differences that I'm missing?

trackleft replied:
No good reason other than trying to keep the name flexible.

trackleft committed:
Merge branch 'main' into issue/2527-add-category-to-citation-ref-display

trackleft committed:
Merge branch 'main' into issue/2527-add-category-to-citation-ref-display
